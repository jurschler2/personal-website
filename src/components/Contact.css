/** 
 * Styling for the Contact component:
 */

/* .contact {
  text-align: center;
  margin: auto;
} */

.contact-header {
  font-family: Kanit;
  font-weight: 300;
  font-size: 115%;
  color: #7b7b87;
  display:  block;
  margin-left: 10%;
  width: 80%;
  text-align: left;
  margin-bottom: 5%;
  margin-top: 2%;
}

.contact-link-container {
  display: inline-block;
  margin-top: 5%;
  width: 80%;
  margin-right:5%;
}

.contact-link {
  display: inline-block;
  width: 20%;
  margin-right: 2%;
  margin-left: 2%;
}

a > .fab.fa-linkedin, a > .fab.fa-angellist, a > .fab.fa-github, a > .fas.fa-paper-plane {
  position: relative;
  font-size: 3rem;
  color:#242337
}

a:hover > .fab.fa-linkedin, a:hover > .fab.fa-angellist, a:hover > .fab.fa-github, a:hover > .fas.fa-paper-plane {
  color:#656573
}

.contact-resume-container {
  display: inline-block;
  margin-top: 5%;
  width: 100%;
  padding: 0 0 6rem 0;
}

.contact-resume {
  display: inline-block;
  margin-top: 5%;
  text-align: left;
  width: 90%;
  margin-left:10%;
  text-decoration: none;
}
.contact-resume-text {
  font-weight: 900;
  font-size: 0.8rem;
  padding: 1rem;
  width: 15rem;
  position: relative;
  top: 2rem;
  background-color: #BDBDC3;
  border-radius: 5px;
  border: solid 2px #39384b;
  color:  #242337;
  text-decoration: none;
}


.contact-resume:hover {
  text-decoration: none;
  color: #656573;
}
.contact-resume-text:hover {
  text-decoration: none;
  color: #656573;
}

@media screen and (min-width: 760px) {

  
  .contact-header {
  font-family: Kanit;
  font-weight: 300;
  font-size: calc(.5rem + 1.5vw);
  color: #7b7b87;
  display:  block;
  margin-left: 10.5%;
  text-align: left;
  margin-bottom: 5%;
  margin-top: 2%;
  }

  .contact-resume-container {
    display: inline-block;
    margin-top: 5%;
    width: 100%;
    padding: 0 0 6rem 0;
  }
  
  .contact-resume {
    display: inline-block;
    margin-top: 5%;
    text-align: left;
    margin-left:10.5%;
    text-decoration: none;
  }
  .contact-resume-text {
    font-weight: 900;
    font-size: 0.8rem;
    padding: 1rem;
    width: 15rem;
    position: relative;
    top: 2rem;
    background-color: #BDBDC3;
    border-radius: 5px;
    border: solid 2px #39384b;
    color:  #242337;
    text-decoration: none;
  }

  
  .contact-resume:hover {
    text-decoration: none;
    color: #656573;
  }
  .contact-resume-text:hover {
    text-decoration: none;
    color: #656573;
  }

  .contact-link-container {
    display: inline-block;
    align-items: left;
    margin-top: 5%;
    margin-right: 50%;
    width: 50%;
  }
  .contact-link {
    display: inline-block;
    align-items: left;
    width: 16%;
    margin-right: 2%;
    margin-left: 2%;
  }
  
  a > .fab.fa-linkedin, a > .fab.fa-angellist, a > .fab.fa-github, a > .fas.fa-paper-plane {
    position: relative;
    left: 3rem;
    font-size: 3rem;
    color:#242337
  }
  
  a:hover > .fab.fa-linkedin, a:hover > .fab.fa-angellist, a:hover > .fab.fa-github, a:hover > .fas.fa-paper-plane {
    color:#656573
  }
}