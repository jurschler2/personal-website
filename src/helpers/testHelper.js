import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export function renderWithRouter(children, initialPath = "") {
  const history = createMemoryHistory();
  if (initialPath) history.push(initialPath);
  return render(<Router history={history}>{children}</Router>);
}
