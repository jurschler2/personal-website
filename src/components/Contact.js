import React from "react";
import "./Contact.css";
import resume from "./JosephUrschlerResume.pdf";

/**
 *  DESCRIPTION:
 *  PROPS: None
 *  FLOW: App => Routes => About
 *  PARENT: Routes
 *  CHILDREN: BulletPoint
 */

function Contact() {
  return (
  <>
    <div className="contact">
      <div className="contact-header">
      Check out my resume and get in contact:
      </div>
      <div className="contact-link-container">
        <div className="contact-link">
          <a href="https://www.linkedin.com/in/jurschler/"><i className="fab fa-linkedin"></i></a>
        </div>
        <div className="contact-link">
          <a href="https://github.com/jurschler2"><i className="fab fa-github"></i></a>
        </div>
        <div className="contact-link">
          <a href="https://angel.co/u/jurschler"><i className="fab fa-angellist"></i></a>
        </div>
        <div className="contact-link">
          <a href="mailto:jurschler25@gmail.com"><i className="fas fa-paper-plane"></i></a>
        </div>
      </div>
      <div className="contact-resume-container">
        <div className="contact-resume">
          <a href={resume} download>
            <p className="contact-resume-text">Download Resume</p>
          </a>
        </div>
      </div>
    </div>
  </>
  );
}

export default Contact;