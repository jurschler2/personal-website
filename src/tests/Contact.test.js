import React from "react";
import Contact from "../components/Contact";
import {renderWithRouter} from "../helpers/testHelper"

it("renders without crashing", () => {
  renderWithRouter(<Contact />);
});

it("matches snapshot", () => {
  const { asFragment } = renderWithRouter(<Contact />);
  expect(asFragment()).toMatchSnapshot();
});