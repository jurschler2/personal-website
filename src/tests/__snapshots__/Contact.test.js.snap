// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="contact"
  >
    <h3>
      Contact
    </h3>
    <div
      class="contact-link-container"
    >
      <a
        href="https://www.linkedin.com/in/jurschler/"
      >
        <i
          class="fab fa-linkedin"
        />
      </a>
      <a
        href="https://github.com/jurschler2"
      >
        <i
          class="fab fa-github"
        />
      </a>
      <a
        href="https://angel.co/u/jurschler"
      >
        <i
          class="fab fa-angellist"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
